Replace user IDs with usernames in Mentions

cyberme0w opened the pull request:
Hi! I wrote a small PR for the feature I proposed in #348. I think the current format (bold+underlined) should be visible enough to stand out.

Cheers!

cyberme0w committed:
Replace user IDs with usernames in Mentions

ayn2op commented on cmd/messages_text.go:
```diff
@@ -111,8 +112,24 @@ func (mt *MessagesText) createHeader(w io.Writer, m discord.Message, isReply boo
+	}
+	return c
+}
+
```
Just use a [strings.Replacer](https://pkg.go.dev/strings#example-NewReplacer) here.

cyberme0w committed:
Convert IDs in single pass with strings.Replacer

ayn2op commented on cmd/messages_text.go:
```diff
@@ -111,8 +112,28 @@ func (mt *MessagesText) createHeader(w io.Writer, m discord.Message, isReply boo
+	}
+	r := strings.NewReplacer(toReplace...)
+	c = r.Replace(c)
+	return c
```
```suggestion
	var toReplace []string
	for _, mention := range m.Mentions {
		toReplace = append(toReplace,
			fmt.Sprintf("<@%s>", mention.User.ID.String()),
			fmt.Sprintf("__**@%s**__", mention.User.Username),
		)
	}

	return strings.NewReplacer(toReplace...).Replace(m.Content)
```

cyberme0w replied:
Maybe it would make sense to return the string-pairs for the replacer, instead of replacing it inside the function. So something like:

```
func getReplacerStringsForUsername() {
  ...
  return toReplace
}
```

and then replace the strings inside `createBody()`.

ayn2op commented on cmd/messages_text.go:
```diff
@@ -111,8 +112,28 @@ func (mt *MessagesText) createHeader(w io.Writer, m discord.Message, isReply boo
+		body = parseIDsToUsernames(m)
+	} else {
+		body = m.Content
+	}
```
```suggestion
	}

```

ayn2op committed:
Update cmd/messages_text.go

cyberme0w committed:
Inline strings.Replacer call

Co-authored-by: Ayyan <119342035+ayn2op@users.noreply.github.com>

cyberme0w committed:
Merge branch 'main' into replace-user-ids-with-names

cyberme0w committed:
Fix markdown to keep replies 'dimmed' after username

The closing bracket should only disable the style of the opening
bracket, and not all styles. Otherwise, nested styles won't work (the
inner style disables the outer style). For instance:

**bold _bold and italic_ bold again**

gets replaced by

[::b]bold [::i]bold and italic[::I] bold again[::B]

instead of

[::b]bold [::i]bold and italic[::-] bold again[::-]
                                   ^^^^^^^^^^^^^^^^
                                       not bold
